File: Lintcode/55.Compare_String.cpp
class Solution {
public:
    /**
     * @param A: A string includes Upper Case letters
     * @param B: A string includes Upper Case letter
     * @return:  if string A contains all of the characters in B return true 
     *           else return false
     */
    bool compareStrings(string A, string B) {
        // write your code here
        if (A.size() < B.size())
            return false;
        int letters[256] = {0};
        for (int i=0; i<A.size(); i++)
            letters[A[i]] += 1;
        for (int i=0; i<B.size(); i++){
            letters[B[i]] -= 1;
            if (letters[B[i]]<0)
                return false;
        }
        return true;
    }
};

File: Lintcode/55.Compare_String.py
class Solution:
    """
    @param A : A string includes Upper Case letters
    @param B : A string includes Upper Case letters
    @return :  if string A contains all of the characters in B return True else return False
    """
    def compareStrings(self, A, B):
        # write your code here
        # 返回一个字典（默认值均为0）
        letters = collections.defaultdict(int)
        for a in A:
            letters[a] += 1
        for b in B:
            if b not in letters:
                return False
            else:
                letters[b] -= 1
                if letters[b] <0:
                    return False
        return True